In [28]:
import pandas as pd
url = "https://raw.githubusercontent.com/dm-fedorov/pandas_basic/master/data/football.csv"
football = pd.read_csv(url)

In [29]:
#Задание 1
round(football.Age.mean())

25

In [30]:
#Задание 2
football.Composure.count()

12897

In [23]:
#Задание 3
round(football.ShortPassing.std(),2)

14.74

In [21]:
#Задание 4
football.Wage.sum()

96954000

In [19]:
#Задание 5
football.Value.min()

0

In [32]:
#Задание 1
round(football[football.Wage > football.Wage.mean()].SprintSpeed.mean(),2)

67.57

In [33]:
#Задание 2
round(football[football.Wage < football.Wage.mean()].SprintSpeed.mean(),2)

62.41

In [37]:
#Задание 3
football[football.Wage==football.Wage.max()].Position
#позиция 0

0    RF
Name: Position, dtype: object

In [40]:
#Задание 4
football[football.Nationality=='Brazil'].Penalties.sum()

22789

In [42]:
#Задание 5
round(football[football.HeadingAccuracy>50].Age.mean(),2)

25.59

In [46]:
#Задание 6
maxComposure=football.Composure.max()
maxReactions=football.Reactions.max()
football[(football.Composure/maxComposure>0.9)&(football.Reactions/maxReactions>0.9)].Age.min()

24

In [51]:
#Задание 7
round(football[football.Age==football.Age.max()].Reactions.mean()-football[football.Age==football.Age.min()].Reactions.mean(),2)


22.64

In [57]:
#Задание 8
football[football.Value>football.Value.mean()].Nationality.max()

'Zimbabwe'

In [62]:
#Задание 9
round(football[(football.Position=="GK")&(football.GKReflexes==football.GKReflexes.max())].Wage.mean()/football[(football.Position=="GK")&(football.GKHandling==football.GKHandling.max())].Wage.mean(),2)

2.77

In [65]:
#Задание 10
round(football[football.Aggression==football.Aggression.max()].ShotPower.mean()/football[football.Aggression==football.Aggression.min()].ShotPower.mean(),2)

2.08

Задание 1: средний возраст 25 лет
Задание 2: 12897
Задание 3: 14.74
Задание 4: 96954000
Задание 5: 0 

Задание 1: 67.57
Задание 2: 62.41
Задание 3: 0
Задание 4: 22789
Задание 5: 25.59
Задание 6: 24
Задание 7: 22.64
Задание 8: 'Zimbabwe'
Задание 9: 2.77
Задание 10: 2.08